rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/EditDynamicParametersWindow.java

```diff
@@ -102,34 +102,36 @@ protected void handleAdditionalInfrastructureFormItems(Map<String, String> selec
     protected List<FormItem> modifyFormItemsAfterCreation(PluginDescriptor focusedInfrastructurePlugin,
             PluginDescriptor focusedPolicyPlugin) {
 
-        List<PluginDescriptor.Field> policyDynamicFields = focusedPolicyPlugin.getConfigurableFields()
-                                                                              .stream()
-                                                                              .filter(PluginDescriptor.Field::isDynamic)
-                                                                              .collect(Collectors.toList());
-
-        List<PluginDescriptor.Field> infrastructureDynamicIFields = focusedInfrastructurePlugin.getConfigurableFields()
-                                                                                               .stream()
-                                                                                               .filter(PluginDescriptor.Field::isDynamic)
-                                                                                               .collect(Collectors.toList());
-        List<PluginDescriptor.Field> allDynamicFields = new LinkedList<>();
-        allDynamicFields.addAll(policyDynamicFields);
-        allDynamicFields.addAll(infrastructureDynamicIFields);
-
+        List<String> policyDynamicFieldFullNames = focusedPolicyPlugin.getConfigurableFields()
+                                                                      .stream()
+                                                                      .filter(PluginDescriptor.Field::isDynamic)
+                                                                      .map(field -> focusedPolicyPlugin.getPluginName() +
+                                                                                    field.getName())
+                                                                      .collect(Collectors.toList());
+        List<String> infrastructureDynamicFieldFullNames = focusedInfrastructurePlugin.getConfigurableFields()
+                                                                                      .stream()
+                                                                                      .filter(PluginDescriptor.Field::isDynamic)
+                                                                                      .map(field -> focusedPolicyPlugin.getPluginName() +
```

Comment:
Should not you use 'focusedInfrastructurePlugin' instead of 'focusedPolicyPlugin'?

Reply:
You are right. It would have been broken if one field of an infrastructure were dynamic. Thanks.